src/platform/plugins/shared/unified_doc_viewer/public/components/observability/traces/components/full_screen_waterfall/index.tsx

```
@@ -97,65 +118,35 @@ export const FullScreenWaterfall = ({
-                      'unifiedDocViewer.observability.traces.fullScreenWaterfall.title',
-                      {
-                        defaultMessage: 'Trace timeline',
-                      }
-                    )}
-                  </h2>
-                </EuiTitle>
-              </EuiFlexItem>
-              <EuiFlexItem grow={1} css={{ alignItems: 'end' }}>
-                <ExitFullScreenButton
-                  onExitFullScreen={onExitFullScreen}
-                  dataTestSubj="unifiedDocViewerObservabilityTracesFullScreenWaterfallExitFullScreenButton"
-                  ariaLabel={i18n.translate(
-                    'unifiedDocViewer.observability.traces.fullScreenWaterfall.exitFullScreen.button',
-                    {
-                      defaultMessage: 'Exit expanded trace timeline',
-                    }
-                  )}
-                />
-              </EuiFlexItem>
-            </EuiFlexGroup>
-            <EuiSpacer size="m" />
-            <EuiFlexGroup>
-              <EuiFlexItem>
-                <EmbeddableRenderer
-                  type="APM_TRACE_WATERFALL_EMBEDDABLE"
-                  getParentApi={getParentApi}
-                  hidePanelChrome
-                />
-              </EuiFlexItem>
-            </EuiFlexGroup>
-          </EuiPanel>
-        </EuiFocusTrap>
-      </EuiOverlayMask>
+    <EuiFlyout
+      session="start"
+      size="m"
+      onClose={onExitFullScreen}
+      aria-labelledby={traceWaterfallTitleId}
+      resizable={true}
+      minWidth={minWidth}
+    >
+      <EuiFlyoutHeader>
+        <EuiTitle size="l">
+          <h2 id={traceWaterfallTitleId}>
+            {i18n.translate('unifiedDocViewer.observability.traces.fullScreenWaterfall.title', {
+              defaultMessage: 'Trace timeline',
+            })}
+          </h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <div ref={embeddableContainerRef}>
+          {scrollElement ? (
```

The conditional render waits for `scrollElement` before mounting the embeddable. Without this, the embeddable would be constructed with `scrollElement = null`, which would break virtualization (embeddables can’t update their initial state).

We added a test to cover the case where the class changes and a white screen is rendered (silent error). Again, I don’t like this approach, but I think it’s necessary.